### PR TITLE
Approve both lenders with retry

### PR DESCRIPTION
**Pull request by shan-57blocks**

### CODE REVIEW

File: packages/huma-widget/src/components/Lend/supplyV2/index.tsx [hunk 1]
```diff
@@ -101,15 +102,8 @@ export function LendSupplyV2({
...
-      if (lenderApprovedJunior && !lenderApprovedSenior) {
-        setSelectedTranche('junior')
-        dispatch(setStep(WIDGET_STEP.ChooseAmount))
-        return
-      }
-
-      if (lenderApprovedSenior && !lenderApprovedJunior) {
-        setSelectedTranche('senior')
-        dispatch(setStep(WIDGET_STEP.ChooseAmount))
+      if (!isUniTranche && (!lenderApprovedJunior || !lenderApprovedSenior)) {
```
Comment (mliu) [inline packages/huma-widget/src/components/Lend/supplyV2/index.tsx:105]: Bugged lenders for our campaign pools will pass the L92 check first and never hit this condition. I think you need to move this check under there.

File: packages/huma-widget/src/components/Lend/solanaSupply/index.tsx [hunk 2]
```diff
@@ -65,13 +66,23 @@ export function SolanaLendSupply({
   const { step, errorMessage } = useAppSelector(selectWidgetState)
   const [selectedTranche, setSelectedTranche] = useState<TrancheType>()
 
+  console.log('isUniTranche', isUniTranche)
```
Comment (mliu) [inline packages/huma-widget/src/components/Lend/solanaSupply/index.tsx:69]: Remove console logs

File: packages/huma-widget/src/components/Lend/solanaSupply/2-ApproveLender.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  isUniTranche: boolean
+  changeTranche: (tranche: TrancheType) => void
+}
+
+export function ApproveLender({
+  poolInfo,
+  isUniTranche,
+  changeTranche,
+}: Props): React.ReactElement | null {
+  const solanChainInfo = SOLANA_CHAIN_INFO[poolInfo.chainId]
```
Comment (mliu) [inline packages/huma-widget/src/components/Lend/solanaSupply/2-ApproveLender.tsx:22]: solanaChainInfo

**Review by mliu — APPROVED** [review]

File: packages/huma-widget/src/components/Lend/supplyV2/index.tsx (hunk 1 above)
Comment (shan-57blocks) [reply packages/huma-widget/src/components/Lend/supplyV2/index.tsx:105]: Updated

File: packages/huma-widget/src/components/Lend/solanaSupply/index.tsx (hunk 2 above)
Comment (shan-57blocks) [reply packages/huma-widget/src/components/Lend/solanaSupply/index.tsx:69]: Updated

File: packages/huma-widget/src/components/Lend/solanaSupply/2-ApproveLender.tsx (hunk 3 above)
Comment (shan-57blocks) [reply packages/huma-widget/src/components/Lend/solanaSupply/2-ApproveLender.tsx:22]: Updated

**Review by mliu — DISMISSED** [review]

**Review by ljiatu — APPROVED** [review]

**Pull request merged**